rfc/wallpapers_v2_network_improvements.md

```diff
@@ -0,0 +1,88 @@
+---
+
+## Summary
+
+The wallpapers feature was introduced on Android and iOS in v98 during Q1 2022. 
+Due to late scope changes, a networking solution was quickly put together
+in allow for delivering wallpapers from a remote source. As part of a
+larger set of improvements planned for Q3 2022, it is intended to revisit
+this implementation to create a more scalable version.
+
+---
+
+## Motivation
+
+The original version of the networking solution consisted of
+adding wallpaper image files to deeply nested directories which defined the properties
+of the wallpaper. For example, an Android wallpaper might be found at:
+```
+<root>/mobile-wallpapers/android/<resolution>/<orientation>/<light/dark>/<wallpaper collection>/<wallpaper name>
+```
+
+This directory structure was confusing and difficult to manage. Copying several versions of a file to different locations
+was painful, even when automated. Automation would require consistent file naming for each new set of wallpapers, which can
+be hard to guarantee. 
+
+Additionally, downloads were started automatically during app startup if a wallpaper was missing.
+
+In general, we would like to achieve the following goals:
+
+- it is easier for product, UX, and ENG to add new wallpapers
+- it is possible to determine important metadata about a wallpaper, like its size or average color
+- download is transparent to the user
+- download is activated by the user
+- require less bandwidth, if possible
+
+---
+
+## Guide-level explanation
+
+Instead of relying on an (undocumented) convention for file paths, wallpaper metadata will be contained in a top-level JSON schema.
+Using a well-known format will help discoverability, and JSON's flexibility should allow for simpler long-term maintenance. 
+It can also provide a more direct path to any required automation. This schema could define things like:
+
+- Android file path(s)
+- iOS file path(s)
+- thumbnail path(s)
+- average color (for text and background coloring purposes)
+- collection grouping for things like seasonal promotions
+- wallpaper availability range
+- wallpaper region availability
+
+An example might look like the following:
+```json
+{
+    "name": "Sunset",
+    "android-file-paths": ["android/path/to/sunset.svg"],
+    "android-thumbnail-path": "android/path/to/sunset-thumbnail.svg",
+    "iOS-file-paths": ["ios/path/to/sunset/high-resolution.pdf", "ios/path/to/sunset/low-resolution.pdf"],
+    "iOS-thumbnail-path": "ios/path/to/sunset-thumbnail.pdf",
+    "average-color": "0xADD8E6",
+    "available-locales": ["en-US", "es-US", "en-CA", "fr-CA"],
+    "availability-range": {
+        "start": "2022-06-27",
+        "end": "2022-09-30"
+    },
+    "collections": ["MR-2022", "FX-100"]
+}
+```
+
+We will likely continue to ship some small number of wallpapers as part of our app binaries. This ensures that at least a few wallpapers are available for onboarding. We could also
```

Comment:
We should include thumbnails for the wallpapers that we ship, but not others. That way, we don't have to download anything on initial app install/launch.